wgpu/src/api/instance.rs

```diff
@@ -2,6 +2,21 @@ use parking_lot::Mutex;
 
 use crate::{dispatch::InstanceInterface, *};
 
+/// WGSL language extensions.
+///
+/// WGSL spec.: <https://www.w3.org/TR/WGSL/#language-extensions-sec>
+#[derive(Debug, Clone, PartialEq, PartialOrd, Ord, Eq, Hash)]
+pub enum WGSLLanguageFeatures {
+    /// <https://www.w3.org/TR/WGSL/#language_extension-readonly_and_readwrite_storage_textures>
+    ReadOnlyAndReadWriteStorageTextures,
+    /// <https://www.w3.org/TR/WGSL/#language_extension-packed_4x8_integer_dot_product>
+    Packed4x8IntegerDotProduct,
+    /// <https://www.w3.org/TR/WGSL/#language_extension-unrestricted_pointer_parameters>
+    UnrestrictedPointerParameters,
+    /// <https://www.w3.org/TR/WGSL/#language_extension-pointer_composite_access>
+    PointerCompositeAccess,
+}
```

Comment:
I think the way we'd want to represent this would be a bitflags

Reply:
Done in https://github.com/gfx-rs/wgpu/pull/6814/commits/4ac1655e3ec914e0404e0280663eb0155ac7064c